depends/lua/src/ldo.c

```diff
@@ -767,14 +767,14 @@ static void f_parser (lua_State *L, void *ud) {
   LClosure *cl;
   struct SParser *p = cast(struct SParser *, ud);
   int c = zgetc(p->z);  /* read first character */
-  if (c == LUA_SIGNATURE[0]) {
-    checkmode(L, p->mode, "binary");
-    cl = luaU_undump(L, p->z, p->name);
-  }
-  else {
+  // if (c == LUA_SIGNATURE[0]) {
```

Comment:
Generally when modifying upstream dependencies that we may upgrade in the future, I would prefer to modify as few lines as possible (kind of how we changed CMakeLists for the libzip/libexpat family of dependencies, although I wouldn't call that a "few" lines...)

Maybe something like
```suggestion
  if (0) { // disabled by DFHack
```

but I think our changes to Lua, including this one, are small enough that I'm not very concerned about maintanability here.